src/class.c

```diff
@@ -1634,22 +1661,27 @@ bool InterfaceDeclaration::isBaseOf(ClassDeclaration *cd, int *poffset)
     {
         BaseClass *b = cd->interfaces[j];
 
-        //printf("\tbase %s\n", b->base->toChars());
-        if (this == b->base)
+        //printf("\tX base %s\n", b->sym->toChars());
+        if (this == b->sym)
         {
             //printf("\tfound at offset %d\n", b->offset);
             if (poffset)
             {
                 *poffset = b->offset;
-                if (j && cd->isInterfaceDeclaration())
+                // The conversion is interface(cd) to interface(this).
+                if (cd->isInterfaceDeclaration())
```

Comment:
Ah, I understand what you ask - if two different type interfaces `IA ia;` and `IB ib` exist, and they are compared by `==`, the "downcast to Object, then compare them" method is used automatically or not - right?

My answer is: Yes. A comparison `ia == ib` (neither IA or IB implements opEquals method) is currently treated as `cast(Object)ia == cast(Object)ib`, and its further translated to `.object.opEquals(cast(Object)ia, cast(Object)ib)`.

The cast from IA/IB to Object will calls `_d_interface_cast()` druntime function (at least it's necessary to handle null interface reference), but the dynamic cast and class hierarchy search will finish immediately. So I can say the total cast cost is enough small.